.gitlab/docker/Dockerfile.spack_build

```diff
@@ -49,3 +49,7 @@ RUN CTEST_XML=$PWD/ctest.xml; \
             --no-compress-output \
             -R tests \
             -no-tests=error"
+
+# Install pika
+RUN spack -e pika_ci build-env $spack_spec -- bash -c "cmake --install ${BUILD_DIR}" && \
+    ./hello_world --pika:bind=none
```

Comment:
Sorry, I was definitely not clear in #906.

We already have unit tests that test the installed pika: `tests.unit.build*`. There are two sets, one that uses the build directory which can always be run and one that uses the install directory which is silently skipped if pika is not installed (because it'd be a constant source of test failures when developing pika).

I think it should be sufficient to simply add `install` to the build targets on line 19.

Reply:
That said, if we add `install` to the targets, you could additionally add a `hello_world` to the test stage, since the examples should get installed.